src/mysql_k8s_helpers.py

```diff
@@ -248,19 +320,59 @@ def _set_group_replication_initial_variables(self) -> None:
             "; ".join(commands), self.cluster_admin_password, self.cluster_admin_user
         )
 
-    def create_custom_config_file(self, report_host: str) -> None:
+    def create_custom_config_file(
+        self,
+        report_host: str,
+        innodb_buffer_pool_size: int,
+        innodb_buffer_pool_chunk_size: int,
+    ) -> None:
         """Create custom configuration file.
 
         Necessary for k8s deployments.
         Raises MySQLCreateCustomConfigFileError if the script gets a non-zero return code.
         """
-        content = ("[mysqld]", f"report_host = {report_host}", "")
+        content = [
+            "[mysqld]",
+            f"report_host = {report_host}",
+            f"innodb_buffer_pool_size = {innodb_buffer_pool_size}",
+        ]
+
+        if innodb_buffer_pool_chunk_size:
+            content.append(f"innodb_buffer_pool_chunk_size = {innodb_buffer_pool_chunk_size}")
+        content.append("")
 
         try:
             self.container.push(MYSQLD_CONFIG_FILE, source="\n".join(content))
         except Exception:
             raise MySQLCreateCustomConfigFileError()
 
+    def offline_mode_and_hidden_instance_exists(self) -> bool:
```

Comment:
After merge, move this method to vm lib!

Reply:
opened [PR](https://github.com/canonical/mysql-operator/pull/125) in the mysql operator to move this to the lib